linkedlist/johnsonparen.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import unicode_literals
+from stack import Stack
+
+
+def paren(str):
+    parenStack = Stack(['end'])
+    for char in unicode(str):
+        if char == '(':
+            parenStack.push(1)
+        elif char == ')':
+            tempchar = parenStack.pop()
+            if tempchar == 'end':
+                return -1
```

Comment:
You can do this, or you could catch the error you know comes from popping an empty stack.  that's probably simpler, and doesn't waste the extra space (even though the space wasted is minimal).